src/widgets/bar/lv_bar.c

```diff
@@ -231,7 +231,9 @@ static void lv_bar_constructor(const lv_obj_class_t * class_p, lv_obj_t * obj)
 
     lv_obj_clear_flag(obj, LV_OBJ_FLAG_CHECKABLE);
     lv_obj_clear_flag(obj, LV_OBJ_FLAG_SCROLLABLE);
+#if LV_WIDGETS_HAS_DEAFULT_VALUE
     lv_bar_set_value(obj, 0, LV_ANIM_OFF);
```

Comment:
I think we should keep. It just ensures that all internal variables are sete correctly.